### PR TITLE
Fixes to stdlib backports

### PR DESCRIPTION
**Pull request by Axel-Naumann**

### CODE REVIEW

**Comment by Axel-Naumann** [comment]
@phsft-bot build also on ubuntu16/classic

File: core/backport/inc/ROOT/memory.hxx [hunk 1]
```diff
@@ -1,4 +1,4 @@
-/// \file ROOT/RMakeUnique.h
+/// \file ROOT/memory.hxx
```
Comment (amadio) [inline core/backport/inc/ROOT/memory.hxx]: Hi @Axel-Naumann, looking at the implementation below, I wonder if we shouldn't try to add a more complete backport, such as the one here: https://isocpp.org/files/papers/N3656.txt. What do you think?

Comment (Axel-Naumann) [reply core/backport/inc/ROOT/memory.hxx]: I'd say not as part of this PR (which aims at sneaking into v6-12 to not have too many release inconsistencies...).

Apart from that I think we should only implement the minimal parts that we need, to reduce the chance that what we provide is not what the standard says.

**Comment by etejedor** [comment]
@Axel-Naumann is this low prio or it can easily go in?

**Comment by Axel-Naumann** [comment]
This is a mixed bag of things. Some is superseded by #1635.

I need to revisit this and start from a fresh PR.

**Pull request closed without merging**